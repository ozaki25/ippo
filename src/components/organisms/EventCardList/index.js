import React from 'react';
import styled from 'styled-components';
import propTypes from 'prop-types';
import { Text } from '@blueprintjs/core';
import EventCard from 'src/components/organisms/EventCard';
import dateFormat from 'src/utils/dateFormat';

const EventCardContainer = styled.div`
  margin: 8px 0;
`;

const EventCardList = ({ events, expand, history }) =>
  events && events.length ? (
    events.map(event => (
      <EventCardContainer key={event.id}>
        <EventCard
          title={event.title}
          eventUrl={event.eventUrl}
          catchMessage={event.catchMessage}
          place={event.place}
          datetime={event.startedAt && `${dateFormat.datetimeJa(event.startedAt)} 〜`}
          interactive
          expand={expand}
          history={history}
        />
      </EventCardContainer>
    ))
  ) : (
    <Text>該当するイベントがありません</Text>
  );
EventCardList.displayName = 'EventCardList';

EventCardList.propTypes = {
  events: propTypes.arrayOf(
    propTypes.shape({
      id: propTypes.oneOfType([propTypes.string, propTypes.number]),
      title: propTypes.string,
      eventUrl: propTypes.string.isRequired,
      catchMessage: propTypes.string,
      place: propTypes.string,
      datetime: propTypes.string,
      interactive: propTypes.bool,
    }),
  ).isRequired,
  expand: propTypes.bool,
  history: propTypes.shape({
    push: propTypes.func,
  }),
};

EventCardList.defaultProps = {
  eventUrl: '',
  expand: false,
  history: null,
};

export default EventCardList;
