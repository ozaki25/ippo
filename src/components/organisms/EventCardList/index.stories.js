import React from 'react';
import EventCardList from '.';

export default {
  title: 'organisms/EventCardList',
};

const event = i => ({
  id: String(i),
  title: 'タイトル',
  eventUrl: 'https://connpass.com/',
  catchMessage: 'イベントの説明です',
  place: '東京都千代田区丸の内',
  startedAt: '2018/12/1 19:30',
});

const props = {
  events: [...[...Array(5)].map((_, i) => event(i))],
};

export const 通常パターン = () => <EventCardList {...props} />;

export const _0件の場合 = () => <EventCardList events={[]} />;
