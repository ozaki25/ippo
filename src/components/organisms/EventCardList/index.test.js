import React from 'react';
import { snapshot } from 'test/helpers';
import EventCardList from '.';

const event = i => ({
  id: String(i),
  title: 'タイトル',
  eventUrl: 'https://connpass.com/',
  catchMessage: 'イベントの説明です',
  place: '東京都千代田区丸の内',
  startedAt: '2018/12/1 19:30',
});

const props = {
  events: [...[...Array(5)].map((_, i) => event(i))],
};

snapshot('EventCardList/nomal', <EventCardList {...props} />);

snapshot('EventCardList/empty', <EventCardList events={[]} />);
