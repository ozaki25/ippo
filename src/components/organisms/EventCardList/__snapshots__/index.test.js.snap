// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: EventCardList/nomal 1`] = `
.c1 {
  min-height: inherit;
}

.c0 {
  margin: 8px 0;
}

<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <EventCardList
    eventUrl=""
    events={
      Array [
        Object {
          "catchMessage": "イベントの説明です",
          "eventUrl": "https://connpass.com/",
          "id": "0",
          "interactive": true,
          "place": "東京都千代田区丸の内",
          "startedAt": "2018/12/1 19:30",
          "title": "タイトル",
        },
        Object {
          "catchMessage": "イベントの説明です",
          "eventUrl": "https://connpass.com/",
          "id": "1",
          "interactive": true,
          "place": "東京都千代田区丸の内",
          "startedAt": "2018/12/1 19:30",
          "title": "タイトル",
        },
        Object {
          "catchMessage": "イベントの説明です",
          "eventUrl": "https://connpass.com/",
          "id": "2",
          "interactive": true,
          "place": "東京都千代田区丸の内",
          "startedAt": "2018/12/1 19:30",
          "title": "タイトル",
        },
        Object {
          "catchMessage": "イベントの説明です",
          "eventUrl": "https://connpass.com/",
          "id": "3",
          "interactive": true,
          "place": "東京都千代田区丸の内",
          "startedAt": "2018/12/1 19:30",
          "title": "タイトル",
        },
        Object {
          "catchMessage": "イベントの説明です",
          "eventUrl": "https://connpass.com/",
          "id": "4",
          "interactive": true,
          "place": "東京都千代田区丸の内",
          "startedAt": "2018/12/1 19:30",
          "title": "タイトル",
        },
      ]
    }
    expand={false}
    history={null}
  >
    <styled.div
      key="0"
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "lastClassName": "c0",
              "rules": Array [
                "
  margin: 8px 0;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="c0"
        >
          <EventCard
            catchMessage="イベントの説明です"
            datetime="2018/12/01 19:30 〜"
            eventUrl="https://connpass.com/"
            expand={false}
            history={null}
            interactive={true}
            place="東京都千代田区丸の内"
            title="タイトル"
          >
            <CustomCard
              expand={false}
              history={null}
              interactive={true}
              url="https://connpass.com/"
            >
              <Styled(Blueprint3.Card)
                elevation={2}
                expand={0}
                onClick={[Function]}
              >
                <StyledComponent
                  elevation={2}
                  expand={0}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bdVaJa",
                        "isStatic": false,
                        "lastClassName": "c1",
                        "rules": Array [
                          "
  min-height: ",
                          [Function],
                          ";
",
                        ],
                      },
                      "displayName": "Styled(Blueprint3.Card)",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bdVaJa",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <Blueprint3.Card
                    className="c1"
                    elevation={2}
                    expand={0}
                    interactive={false}
                    onClick={[Function]}
                  >
                    <div
                      className="bp3-card bp3-elevation-2 c1"
                      expand={0}
                      onClick={[Function]}
                    >
                      <h3
                        className="bp3-heading"
                      >
                        タイトル
                      </h3>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          概要：イベントの説明です
                        </div>
                      </Blueprint3.Text>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          場所：東京都千代田区丸の内
                        </div>
                      </Blueprint3.Text>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          日時：2018/12/01 19:30 〜
                        </div>
                      </Blueprint3.Text>
                    </div>
                  </Blueprint3.Card>
                </StyledComponent>
              </Styled(Blueprint3.Card)>
            </CustomCard>
          </EventCard>
        </div>
      </StyledComponent>
    </styled.div>
    <styled.div
      key="1"
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "lastClassName": "c0",
              "rules": Array [
                "
  margin: 8px 0;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="c0"
        >
          <EventCard
            catchMessage="イベントの説明です"
            datetime="2018/12/01 19:30 〜"
            eventUrl="https://connpass.com/"
            expand={false}
            history={null}
            interactive={true}
            place="東京都千代田区丸の内"
            title="タイトル"
          >
            <CustomCard
              expand={false}
              history={null}
              interactive={true}
              url="https://connpass.com/"
            >
              <Styled(Blueprint3.Card)
                elevation={2}
                expand={0}
                onClick={[Function]}
              >
                <StyledComponent
                  elevation={2}
                  expand={0}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bdVaJa",
                        "isStatic": false,
                        "lastClassName": "c1",
                        "rules": Array [
                          "
  min-height: ",
                          [Function],
                          ";
",
                        ],
                      },
                      "displayName": "Styled(Blueprint3.Card)",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bdVaJa",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <Blueprint3.Card
                    className="c1"
                    elevation={2}
                    expand={0}
                    interactive={false}
                    onClick={[Function]}
                  >
                    <div
                      className="bp3-card bp3-elevation-2 c1"
                      expand={0}
                      onClick={[Function]}
                    >
                      <h3
                        className="bp3-heading"
                      >
                        タイトル
                      </h3>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          概要：イベントの説明です
                        </div>
                      </Blueprint3.Text>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          場所：東京都千代田区丸の内
                        </div>
                      </Blueprint3.Text>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          日時：2018/12/01 19:30 〜
                        </div>
                      </Blueprint3.Text>
                    </div>
                  </Blueprint3.Card>
                </StyledComponent>
              </Styled(Blueprint3.Card)>
            </CustomCard>
          </EventCard>
        </div>
      </StyledComponent>
    </styled.div>
    <styled.div
      key="2"
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "lastClassName": "c0",
              "rules": Array [
                "
  margin: 8px 0;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="c0"
        >
          <EventCard
            catchMessage="イベントの説明です"
            datetime="2018/12/01 19:30 〜"
            eventUrl="https://connpass.com/"
            expand={false}
            history={null}
            interactive={true}
            place="東京都千代田区丸の内"
            title="タイトル"
          >
            <CustomCard
              expand={false}
              history={null}
              interactive={true}
              url="https://connpass.com/"
            >
              <Styled(Blueprint3.Card)
                elevation={2}
                expand={0}
                onClick={[Function]}
              >
                <StyledComponent
                  elevation={2}
                  expand={0}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bdVaJa",
                        "isStatic": false,
                        "lastClassName": "c1",
                        "rules": Array [
                          "
  min-height: ",
                          [Function],
                          ";
",
                        ],
                      },
                      "displayName": "Styled(Blueprint3.Card)",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bdVaJa",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <Blueprint3.Card
                    className="c1"
                    elevation={2}
                    expand={0}
                    interactive={false}
                    onClick={[Function]}
                  >
                    <div
                      className="bp3-card bp3-elevation-2 c1"
                      expand={0}
                      onClick={[Function]}
                    >
                      <h3
                        className="bp3-heading"
                      >
                        タイトル
                      </h3>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          概要：イベントの説明です
                        </div>
                      </Blueprint3.Text>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          場所：東京都千代田区丸の内
                        </div>
                      </Blueprint3.Text>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          日時：2018/12/01 19:30 〜
                        </div>
                      </Blueprint3.Text>
                    </div>
                  </Blueprint3.Card>
                </StyledComponent>
              </Styled(Blueprint3.Card)>
            </CustomCard>
          </EventCard>
        </div>
      </StyledComponent>
    </styled.div>
    <styled.div
      key="3"
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "lastClassName": "c0",
              "rules": Array [
                "
  margin: 8px 0;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="c0"
        >
          <EventCard
            catchMessage="イベントの説明です"
            datetime="2018/12/01 19:30 〜"
            eventUrl="https://connpass.com/"
            expand={false}
            history={null}
            interactive={true}
            place="東京都千代田区丸の内"
            title="タイトル"
          >
            <CustomCard
              expand={false}
              history={null}
              interactive={true}
              url="https://connpass.com/"
            >
              <Styled(Blueprint3.Card)
                elevation={2}
                expand={0}
                onClick={[Function]}
              >
                <StyledComponent
                  elevation={2}
                  expand={0}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bdVaJa",
                        "isStatic": false,
                        "lastClassName": "c1",
                        "rules": Array [
                          "
  min-height: ",
                          [Function],
                          ";
",
                        ],
                      },
                      "displayName": "Styled(Blueprint3.Card)",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bdVaJa",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <Blueprint3.Card
                    className="c1"
                    elevation={2}
                    expand={0}
                    interactive={false}
                    onClick={[Function]}
                  >
                    <div
                      className="bp3-card bp3-elevation-2 c1"
                      expand={0}
                      onClick={[Function]}
                    >
                      <h3
                        className="bp3-heading"
                      >
                        タイトル
                      </h3>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          概要：イベントの説明です
                        </div>
                      </Blueprint3.Text>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          場所：東京都千代田区丸の内
                        </div>
                      </Blueprint3.Text>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          日時：2018/12/01 19:30 〜
                        </div>
                      </Blueprint3.Text>
                    </div>
                  </Blueprint3.Card>
                </StyledComponent>
              </Styled(Blueprint3.Card)>
            </CustomCard>
          </EventCard>
        </div>
      </StyledComponent>
    </styled.div>
    <styled.div
      key="4"
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "lastClassName": "c0",
              "rules": Array [
                "
  margin: 8px 0;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="c0"
        >
          <EventCard
            catchMessage="イベントの説明です"
            datetime="2018/12/01 19:30 〜"
            eventUrl="https://connpass.com/"
            expand={false}
            history={null}
            interactive={true}
            place="東京都千代田区丸の内"
            title="タイトル"
          >
            <CustomCard
              expand={false}
              history={null}
              interactive={true}
              url="https://connpass.com/"
            >
              <Styled(Blueprint3.Card)
                elevation={2}
                expand={0}
                onClick={[Function]}
              >
                <StyledComponent
                  elevation={2}
                  expand={0}
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bdVaJa",
                        "isStatic": false,
                        "lastClassName": "c1",
                        "rules": Array [
                          "
  min-height: ",
                          [Function],
                          ";
",
                        ],
                      },
                      "displayName": "Styled(Blueprint3.Card)",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bdVaJa",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                      Symbol(Symbol.hasInstance): [Function],
                    }
                  }
                  forwardedRef={null}
                  onClick={[Function]}
                >
                  <Blueprint3.Card
                    className="c1"
                    elevation={2}
                    expand={0}
                    interactive={false}
                    onClick={[Function]}
                  >
                    <div
                      className="bp3-card bp3-elevation-2 c1"
                      expand={0}
                      onClick={[Function]}
                    >
                      <h3
                        className="bp3-heading"
                      >
                        タイトル
                      </h3>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          概要：イベントの説明です
                        </div>
                      </Blueprint3.Text>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          場所：東京都千代田区丸の内
                        </div>
                      </Blueprint3.Text>
                      <Blueprint3.Text>
                        <div
                          className=""
                        >
                          日時：2018/12/01 19:30 〜
                        </div>
                      </Blueprint3.Text>
                    </div>
                  </Blueprint3.Card>
                </StyledComponent>
              </Styled(Blueprint3.Card)>
            </CustomCard>
          </EventCard>
        </div>
      </StyledComponent>
    </styled.div>
  </EventCardList>
</MockRouter>
`;
